pkg/cloud/gcp/actuators/machineset/controller.go

```diff
@@ -175,6 +176,29 @@ func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, e
 	machineSet.Annotations[labelsKey] = mapiutil.MergeCommaSeparatedKeyValuePairs(
 		fmt.Sprintf("kubernetes.io/arch=%s", util.CPUArchitecture(providerConfig.MachineType)),
 		machineSet.Annotations[labelsKey])
+
+	// When upgrading from 4.12 on GCP marketplace, the MachineSets refer to images that do not support UEFI & shielded VMs.
+	// However, GCP defaulted to shielded VMs sometime between 4.12 and 4.13.
+	// Therefore, we should disable the shielded instance config in the MachineSet's template, so that new Machines created from it will boot.
+	uefiCompatible, err := isUEFICompatible(gceService, providerConfig)
+	if err != nil {
+		return ctrl.Result{}, mapierrors.InvalidMachineConfiguration("error fetching disk information: %v", err)
```

Comment:
It could be transient, though I'm not sure when it would resolve.

The error type probably needs to be different in any case. Invalid configuration was the closest of our existing types, but really the invalidity would come from us/our code rather than the user.